chalk-engine/src/logic.rs

```diff
@@ -117,12 +117,15 @@ impl<I: Interner> Forest<I> {
     pub(super) fn any_future_answer(
         &self,
         table: TableIndex,
-        answer: AnswerIndex,
+        mut answer_index: AnswerIndex,
         mut test: impl FnMut(&Substitution<I>) -> bool,
     ) -> bool {
-        if let Some(answer) = self.tables[table].answer(answer) {
+        while let Some(answer) = self.tables[table].answer(answer_index) {
```

Comment:
We should write a better comment here, I think. Maybe something like:

First, check the cached answers, starting from `answer_index`. Then check whether the partial values for strands, as they may produce future cached answers.